### DISCUSSION TITLE
What is the relationship with ROCm and opencl-amd in AUR of Arch Linux?

### DISCUSSION THREAD

**Discussion by HubKing** [ROCm/ROCm#1923] [Q&A] [answered]
On Arch-based distributions, there is no "ROCm" in the repertoires, and but only [opencl-amd](https://aur.archlinux.org/packages/opencl-amd). Since I don't know how/want to build and manually install ROCm, I just installed opencl-amd. My only purpose of doing this is to use HIP rendering in Blender. 

After installing opencl-amd (and doing some weird things), HIP rendering works in Blender, but I am not sure what the difference is. Is it any better to use this ROCm instead of opencl-amd?

**Comment by sofiageo** [comment, marked as answer]
`opencl-amd` and `opencl-amd-dev` is just the Ubuntu packages of ROCm extracted in `/opt/rocm` directory. (source: I maintain both AUR packages)

There is also a new official package ~~[rocm-opencl-runtime](https://archlinux.org/packages/?name=rocm-opencl-runtime) which should provide the same functionality - but I haven't personally used it, and I'm not sure if it requires any more packages to get the full ROCm capabilities.~~

edit: I did a little digging myself, I think these two packages `rocm-hip-sdk` and `rocm-opencl-sdk` should be enough for HIP/OpenCL support: https://wiki.archlinux.org/title/GPGPU#ROCm

**Reply by HubKing** [reply]
If all one wants is HIP rendering in Blender, does he need to install rocm-hip-sdk, not rocm-hip-runtime?

**Reply by sofiageo** [reply]
I think it depends on what version of Blender you are using, I think the one you download from the blender website can work with only the hip runtime, but the one from the arch repositories might need the hip-sdk. Or at least that's what some people have reported. When in doubt, just use rocm-hip-sdk which includes everything (but is also larger)
